src/class/Server/Server.cpp

```diff
@@ -13,7 +13,18 @@ Server::Server(const IServerConf *conf)
 	std::list< const IListen * >::iterator end = tmpListeners.end();
 	for (; it != end; it++)
 	{
-		this->_sockets.push_back(new Socket(*it));
+		Socket *new_socket;
+		try
+		{
+			new_socket = new Socket(*it);
```

Comment:
why are the _sockets a Socket*?
can't they be: std::vector< Socket > or std::list< Socket >